### PR TITLE
[feature] #2922: Simplify JSON deserialization

### LINKED ISSUE #2922: [suggestion] Simplify JSON deserialization further
### Feature request

Consider extending the shorthand syntax to include the following case: 
```json
"Register": {
	"Identifiable": {
		"NewAssetDefinition": {
			"id": "sapporo#neverland",
			"value_type": "Quantity",
			"mintable": "Infinitely",
			"metadata": {}
		}
	}
}
```

where the Identifiable is obvious from internal representations. We can't (at this point in time), register anything that isn't part of the `IdentifiableBox` `enum`. So we could extend the idea due to @Arjentix to shorten the syntax: 

```json
"Register": {
	"NewAssetDefinition": {
		"id": "sapporo#neverland",
		"value_type": "Quantity",
		"mintable": "Infinitely"
	}
}
```

### Motivation

Shorter syntax ⇒ less space for large genesis blocks ⇒ less work. 

### Who can help?

@appetrosyan

### PR DESCRIPTION
**Pull request by Erigara**
### Description of the Change

Omit `Identifiable`, `Id`, `object(_id)` fields in json deserialization.

### Issue

Closes #2922 

### Benefits

Genesis take smaller amount of space, shorter to write.

### Possible Drawbacks

None.

### Usage Examples

Field `Identifiable` is omitted:

```diff
         {
           "Register": {
-            "Identifiable": {
              "NewDomain": {
                "id": "wonderland",
                "logo": null,
                "metadata": {
                  "key": {
                    "String": "value"
                  }
                 }
-               }
             }
         }
```

```diff
         {
           "Register": {
-            "Identifiable": {
              "NewAccount": {
                "id": "alice@wonderland",
                "signatories": [
                  "ed01207233bfc89dcbd68c19fde6ce6158225298ec1131b6a130d1aeb454c1ab5183c0"
                ],
                "metadata": {
                  "key": {
                    "String": "value"
                  }
                 }
-               }
             }
           }
```

Fields `Id`, `object` and `object_id` are omitted:

```diff
         {
           "Mint": {
-            "object": {
              "U32": 13
-            },
             "destination_id": {
-              "Id": {
                "AssetId": "rose##alice@wonderland"
-              }
             }
           }
         }
```

Field `inner` of `NewRole` is omitted:

```diff
            "NewRole": {
-            "inner": {
              "id": "USER_METADATA_ACCESS",
              "permissions": [
                {
                  "definition_id": "can_remove_key_value_in_user_metadata",
                  "params": {
                    "account_id": {
                      "AccountId": "alice@wonderland"
                    }
                  }
                },
                {
                  "definition_id": "can_set_key_value_in_user_metadata",
                  "params": {
                    "account_id": {
                      "AccountId": "alice@wonderland"
                    }
                  }
                }
              ]
-            }
            }
```

### CODE REVIEW

**Review by mversic — APPROVED** [review]
love it

File: configs/peer/genesis.json [hunk 1]
```diff
@@ -4,341 +4,281 @@
...
-                "inner": {
-                  "id": "staff_that_does_stuff_in_genesis",
-                  "permissions": [
-                    {
-                      "definition_id": "allowed_to_do_stuff",
-                      "params": {}
-                    }
-                  ]
-                }
+            "NewRole": {
```
Comment (appetrosyan) [inline configs/peer/genesis.json:273]: Consider removing the "inner" layer as well.

File: tools/parity_scale_decoder/samples/trigger.json [hunk 2]
```diff
@@ -5,15 +5,9 @@
...
-                        "object": {
-                            "Numeric": {
-                                "U32": 1
-                            }
-                        },
+                        "U32": 1,
                         "destination_id": {
-                            "Id": {
-                                "AssetId": "rose##alice@wonderland"
-                            }
```
Comment (appetrosyan) [inline tools/parity_scale_decoder/samples/trigger.json:16]: Given that it's unambiguously an AssetId, we could simplify the syntax further to just equal the Id.

**Review by appetrosyan — APPROVED** [review]

**Comment by appetrosyan** [comment]
Small side not, is it possible to add an example of granting a role in the genesis as well? We want to have an example of almost everything if we can.

File: tools/parity_scale_decoder/samples/trigger.json (hunk 2 above)
Comment (Erigara) [reply tools/parity_scale_decoder/samples/trigger.json:16]: `Mint` isi is also available for `PublicKey` and `SignatureCheckCondition` so i don't think we can simplify further.

**Review by mversic — APPROVED** [review]

**Review by appetrosyan — APPROVED** [review]

**Pull request merged**